### DISCUSSION TITLE
Secret variable is not masked in “Task Instance Details”

### DISCUSSION THREAD

**Discussion by mkalamcargurus** [apache/airflow#30648] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

I set a secret as a variable and Airflow recognizes it as such and masks the value when trying to view the variables and when it is referenced in the logs. However, you can see the value when you click on “Task Instance Details” for a DAG that is using the variable. 
MWAA 2.4.3

### What you think should happen instead

The secret value should be masked.

### How to reproduce

Add secret as a variable, name it password.
Use the secret in a DAG, our example we were using it in the DAG overrides block. 
Run an instance of the given DAG and navigate to the overrides section under 'Task Instance Details"

### Operating System

MWAA

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon | 6.0.0 | Amazon integration (including Amazon Web Services (AWS)).
apache-airflow-providers-celery | 3.0.0 | Celery
apache-airflow-providers-common-sql | 1.2.0 | Common SQL Provider
apache-airflow-providers-ftp | 3.1.0 | File Transfer Protocol (FTP)
apache-airflow-providers-http | 4.0.0 | Hypertext Transfer Protocol (HTTP)
apache-airflow-providers-imap | 3.0.0 | Internet Message Access Protocol (IMAP)
apache-airflow-providers-postgres | 5.2.2 | PostgreSQL
apache-airflow-providers-sqlite | 3.2.1 | SQLite

### Deployment

Amazon (AWS) MWAA

### Deployment details

mwaa

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by maahir22** [comment]
If triaged, I'd like to work on this issue. Please /assign.

**Comment by potiuk** [comment]
Feel free. You will be triaging it while working on it. If we find out the issue is not there, we will close it.

**Comment by mkalamcargurus** [comment]
I'd be happy to help provide more info/screenshots if needed.

**Comment by hussein-awala** [comment]
@mkalamcargurus can you check if you do the same thing as #30039 ?

**Comment by hussein-awala** [comment]
Task Instance Details contains the operator arguments, so if you load the secret in the dag script outside the method execute (which is not recommended and considered as a bad practice), then you provide it as an argument for the operator, the operator will process it as a normal argument.
if it's the case, you can use jinja instead and load it in runtime:
```python
import datetime

from airflow import DAG
from airflow.models.baseoperator import BaseOperator
from airflow.models import Variable


class MyOperator(BaseOperator):
    template_fields = ("some_arg",)

    def __init__(self, some_arg, **kwargs):
        super().__init__(**kwargs)
        self.some_arg = some_arg

    def execute(self, context):
        print(self.some_arg)


with DAG(
    'secret_masking_test',
    schedule_interval=None,
    catchup=False,
    start_date=datetime.datetime(2018, 1, 1)
) as dag:
    bad_task = MyOperator(task_id="bad_task", some_arg=Variable.get("password"))
    good_task = MyOperator(task_id="good_task", some_arg="{{ var.value.password }}")
```
bad_task:
![image](https://user-images.githubusercontent.com/21311487/228974541-ae8cbccb-dc01-49cf-8f9b-723b1f7620e2.png)

good_task:
![image](https://user-images.githubusercontent.com/21311487/228974683-d8aed640-9e7b-4f8c-b000-f1b1971b5fcb.png)
and it's loaded just before the method execute, and you can use it inside the method without any problem.

If your problem is different, please provide a reproducible example.

**Comment by potiuk** [comment]
Converted into disussion - if there is more input, please provide one @mkalamcargurus

**Comment by AliAzG** [comment]
I have the same problem with Airflow version 2.6.3, where I pass my credentials in a bash command, and in Task Instance Details in section **Attribute: bash_command** the values I did **mask_secret** on them are not masked. In Rendered Template it's ok though.

Is there any solution for this issue?

Thanks

**Reply by potiuk** [reply]
Look at the changelog of Airflow and CVEs reported (set to . Not sure if it solves your problem but there were some CVEs reported (We announce them at users@apache.org archive here: https://lists.apache.org/list.html?users@airflow.apache.org - also there are vulnerability databases you can browse. If we fix similar issue we issue CVE advisories to our users that they should be following - so you should check there. 

If after reviewing those and (most likely) upgrading to the latest Airflow you will see that your scenario is not covered, please report the reproduction scenario following our security policy and "responsible disclosure" policy https://github.com/apache/airflow/security/policy . Discussing potential security issue details should not happen in public issue/discssion - follow the security policy on that (but only after upgrading). You will find that there were a lot of CVEs sent in the recent months advising our users mostly to upgrade to latest versions that address a number of vulnerabilities like this one.
